core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStore.java

```diff
@@ -272,10 +272,11 @@ private void createSchemaTopic() throws StoreInitializationException {
   }
 
   static List<String> brokersToEndpoints(List<Broker> brokers) {
-    List<String> endpoints = new LinkedList<String>();
+    List<String> endpoints = new LinkedList<>();
     for (Broker broker : brokers) {
       for (EndPoint ep : JavaConversions.asJavaCollection(broker.endPoints())) {
-        endpoints.add(ep.connectionString());
+        String hostport = ep.host() == null ? ":" + ep.port() : Utils.formatAddress(ep.host(), ep.port());
+        endpoints.add(ep.securityProtocol() + "://" + hostport);
```

Comment:
The decision to use a separate `security.protocol` config for the clients was done before my time, but it makes sense from the perspective that only one security protocol is supported. The fact that we don't validate it makes it a bit confusing and it may make sense to add a consistency check in the future.